payu/envmod.py

```diff
@@ -104,15 +104,12 @@ def lib_update(bin_path, lib_name):
     slibs = ldd_output.split('\n')
 
     for lib_entry in slibs:
-        if lib_name in lib_entry:
+        if lib_name in lib_entry and 'spack' not in lib_entry:
+            # Only load enviroment modules available - assuming spack built libs includes 'spack' in full path of library
             lib_path = lib_entry.split()[2]
 
             # pylint: disable=unbalanced-tuple-unpacking
             mod_name, mod_version = fsops.splitpath(lib_path)[2:4]
 
             module('unload', mod_name)
-            module('load', os.path.join(mod_name, mod_version))
-            return '{0}/{1}'.format(mod_name, mod_version)
-
-    # If there are no libraries, return an empty string
-    return ''
```

Comment:
I can see that the return value is no longer used here, or anywhere else in the code. I'm not sure we should remove that functionality though. It doesn't do any harm, and it may be something that @marshallward would expect to be available, and I can see the utility in returning the result of this sort of inspection.

In addition, if we wanted to test this new functionality we should split out the logic for finding a `lib_name` in the `ldd` output, and return `mod_name, mod_version` from that function.

Then we could add a `pytest` test with some pre-generated `ldd` output that included `spack` libraries and check we get the correct return values.

Reply:
Ok, I've reverted those changes and added in a couple of tests. I've also pulled the changes recommended here: https://github.com/payu-org/payu/issues/341#issuecomment-1641192337